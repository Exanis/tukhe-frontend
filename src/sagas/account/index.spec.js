import axios from 'axios';
import watcherSaga, {
    loadAccountSaga,
    deleteAccountSaga,
    fetchDeleteAccount,
    fetchAccountsList
} from './index';

it('Should have a watcher saga as expected', () => {
    const gen = watcherSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should load accounts as expected', () => {
    const gen = loadAccountSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next({
        data: []
    })).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should delete accounts as expected', () => {
    const gen = deleteAccountSaga({
        payload: 'test'
    });
    const request = gen.next();

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should prepare ajax requests as expected', () => {
    expect(fetchAccountsList()).toMatchSnapshot();
    expect(fetchDeleteAccount('test')).toMatchSnapshot();
});

it('Should handle error on delete', () => {
    const gen = deleteAccountSaga({
        payload: 'test'
    });

    expect(gen.next()).toMatchSnapshot();
    gen.throw('error');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should handle error on load accounts as expected', () => {
    const gen = loadAccountSaga();

    expect(gen.next()).toMatchSnapshot();
    gen.throw('error');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});
