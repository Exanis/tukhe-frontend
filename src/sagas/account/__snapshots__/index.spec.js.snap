// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should delete accounts as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should delete accounts as expected 2`] = `Promise {}`;

exports[`Should delete accounts as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": undefined,
        "type": "accounts/load/list",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should handle error on delete 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should handle error on delete 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should handle error on load accounts as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should handle error on load accounts as expected 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should have a watcher saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should have a watcher saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should have a watcher saga as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should load accounts as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should load accounts as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Array [],
        "type": "account/setAccounts",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should prepare ajax requests as expected 1`] = `Promise {}`;

exports[`Should prepare ajax requests as expected 2`] = `Promise {}`;
